pkg/compact/compact.go

```diff
@@ -904,7 +1004,9 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 			go func() {
 				defer wg.Done()
 				for g := range groupChan {
+					c.grouper.StartGroup(g)
```

Comment:
Why we need that BTW? Do we ever plan/group and compact in the same time? 🤔

Reply:
The reason for this is compacting methods return shouldRerunGroup as true and when i first tested this it seemed that after one group completed the planner would see blocks that are currently being compacted and add groups for them. So start group keeps track of whats currently being compacted and when skip those here https://github.com/thanos-io/thanos/pull/3807/files#diff-be21b0381ac14ee139355d057b390aa69ff338fe1c748ae8a475ce46f3583575R311

Reply:
After looking into this more, this change I think is still necessary. Say that compaction iterations are set to happen every 30 minutes but a large tenant has compaction runs that take more than 30 minutes. Yes you can increase the compaction frequency but this change would prevent a new iteration trying to compact blocks that are already being compacted.